### DISCUSSION TITLE
Translations for products randomly disappear

### DISCUSSION THREAD

**Discussion by agsmartwolf** [swellstores/community#222] [Help] [answered]
Hi guys,

Sometimes product descriptions in other languages disappear except in English.
It started long ago, but I thought it would be fixed by a major update.
However, translations for some of the products (random products) are removed. English product description persists but other ones are removed and I should translate them again and again.

**Comment by SusannaSwell** [comment]
hello @agsmartwolf , tried to reproduce this on my end, but without success. Also can't find any other reports about this issue. 
Could you please post a link to one of products like you described (or API product data), so our team could investigate this. Also any additional information can be useful here, maybe you're using export/import of product to edit them, or you're editing in the dashboard?

**Reply by agsmartwolf** [reply]
We edit in the dashboard. I'd like to give you more details and I tried to reproduce it by some manipulations, but I couldn't.

Examples of products:
https://smart-wolf.swell.store/admin/products/642befff04c1620012a6e445

https://smart-wolf.swell.store/admin/products/63f89420750bbf00123dbf29 (all products in the category "LickiMat")


Now we manually translated again.

**Reply by SusannaSwell** [reply]
Thank you for the details, we will be investigating further. I will get back to you once we get an update.

**Reply by SusannaSwell** [reply]
We checked this product's update history and can't see if the localized description has ever been removed.
https://smart-wolf.swell.store/admin/console?get/events//-(data.id:642befff04c1620012a6e445)

2023-04-04T10:17:53.228Z added russian description
2023-04-18T09:25:30.157Z russian description was updated

Can you please provide more information where descriptions disappear: admin panel only, or storefront also? A network log at this moment can be extremely helpful.

**Reply by agsmartwolf** [reply]
We saw disappeared translations on the dashboard. I don't see translations on the admin panel right now.
For this product: https://smart-wolf.swell.store/admin/products/63f89420750bbf00123dbf29


HAR file logs:
[smart-wolf.swell.store.har.zip](https://github.com/swellstores/community/files/11382372/smart-wolf.swell.store.har.zip)

Missed translations:
<img width="1281" alt="Screenshot 2023-05-03 at 16 17 43" src="https://user-images.githubusercontent.com/109105178/235913962-89e752a3-6a79-4a5c-9309-c1f68bd52fc7.png">

<img width="1229" alt="Screenshot 2023-05-03 at 16 18 04" src="https://user-images.githubusercontent.com/109105178/235913933-3eda0d03-8fd9-4242-adfe-1f1f73f55749.png">

**Reply by agsmartwolf** [reply]
I have returned English translations for this product in the runtime too.

Request and returned response:

<img width="1878" alt="Screenshot 2023-05-03 at 16 27 05" src="https://user-images.githubusercontent.com/109105178/235915393-93f66dff-5415-4af8-8a64-16ee554b2c93.png">

<img width="1905" alt="Screenshot 2023-05-03 at 16 25 47" src="https://user-images.githubusercontent.com/109105178/235915053-73c32a77-21e1-4a7b-8a0c-6cdfa3cf2057.png">

**Reply by SusannaSwell** [reply]
Thank you for the additional details, we will be investigating further.

**Reply by SusannaSwell** [reply]
We checked all available logs, but can't see a record where 'ru' locale was removed.

From the event log, we see that the "ru" locale was filled in for this product (https://smart-wolf.swell.store/admin/console?get/products/-63f89420750bbf00123dbf29) at `2023-04-18T09:46:00.921Z`

There is one event in the event logs after April 18th and it is without any content changes: `2023-04-30T07:48:36.083Z`
We can't identify any changes for this product at 04.30.

Maybe you can recall if there was any change on 04.30? this change could potentially delete 'ru' locale, but it is not identified in our logs

**Reply by logeshswell** [reply, marked as answer]
@agsmartwolf We will create a bug ticket internally to investigate it further to identify the root cause and fix it at the earliest.

**Reply by agsmartwolf** [reply]
Thank you. 
I'm looking forward to results.
I can't determine what exactly actions were made to effect to the translations.
However translations are dissapeared for some of products more often than for other ones.

I should recreate them manually again now because I don't have translations on the production right now.

**Reply by agsmartwolf** [reply]
> We checked all available logs, but can't see a record where 'ru' locale was removed.
> 
> From the event log, we see that the "ru" locale was filled in for this product (https://smart-wolf.swell.store/admin/console?get/products/-63f89420750bbf00123dbf29) at `2023-04-18T09:46:00.921Z`
> 
> There is one event in the event logs after April 18th and it is without any content changes: `2023-04-30T07:48:36.083Z` We can't identify any changes for this product at 04.30.
> 
> Maybe you can recall if there was any change on 04.30? this change could potentially delete 'ru' locale, but it is not identified in our logs

I've talked to my employees. On 30 of April everything was untouchable at all. So we didn't have any deployments and any changes on admin dashboard,
